lib/src/parsers/utilities/parser-utility.ts

```diff
@@ -462,12 +469,43 @@ export function getTargetDeclarationFromTypeReference(
     }
   }
   const targetDeclaration = declarations[0];
+  if (
+    TypeGuards.isInterfaceDeclaration(targetDeclaration) &&
+    targetDeclaration.getIndexSignatures().length > 0
+  ) {
+    throw new Error(
+      `indexed types are not supported (offending type: ${targetDeclaration.getName()})`
+    );
+  }
+  if (TypeGuards.isTypeAliasDeclaration(targetDeclaration)) {
+    const typeNode = targetDeclaration.getTypeNodeOrThrow();
+    if (
+      TypeGuards.isTypeLiteralNode(typeNode) &&
+      typeNode.getIndexSignatures().length > 0
+    ) {
+      throw new Error(
+        `indexed types are not supported (offending type: ${targetDeclaration.getName()})`
+      );
+    }
+  }
   if (
     TypeGuards.isInterfaceDeclaration(targetDeclaration) ||
     TypeGuards.isTypeAliasDeclaration(targetDeclaration)
   ) {
     return targetDeclaration;
   }
+  if (TypeGuards.isEnumDeclaration(targetDeclaration)) {
+    throw new Error(
+      `enums are not supported (offending type: ${targetDeclaration.getName()})`
+    );
+  }
+  if (TypeGuards.isEnumMember(targetDeclaration)) {
+    throw new Error(
+      `enums are not supported (offending type: ${targetDeclaration
+        .getParent()
```

Comment:
What's `parent` for and why only in enums?

Reply:
The parent of an enum member is the enum type itself. For example when the type is `SomeEnum.A`, we want to show "offending type: SomeEnum".